### ISSUE TITLE
Update rpm dependencies file for newer CXXABI requirement

### ISSUE THREAD

**Issue by tazerdev** [microsoft/vscode#115784]
VS Code Version: 1.53.0
OS: RHEL7.9

Error:
The terminal process failed to launch: A native exception occurred during launch (/lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /usr/share/code/resources/app/node_modules.asar.unpacked/node-pty/build/Release/pty.node)).

```
$ strings /lib64/libstdc++.so.6 | grep CXXABI
CXXABI_1.3
CXXABI_1.3.1
CXXABI_1.3.2
CXXABI_1.3.3
CXXABI_1.3.4
CXXABI_1.3.5
CXXABI_1.3.6
CXXABI_1.3.7
CXXABI_TM_1
```

**Comment by gjsjohnmurray** [comment]
See Breaking Change section at https://code.visualstudio.com/updates/v1_53#_performance-improvements

**Comment by gjsjohnmurray** [comment]
Better link https://code.visualstudio.com/updates/v1_53#_electron-11-update

**Comment by dcourtois** [comment]
Same issue here. Unfortunately CentOS 7 is still wildly use in some areas, and the links pointed to by @gjsjohnmurray explains the cause, but unfortunately doesn't provide any workaround.
Is there any workaround known at this point ? This is a pretty important issue, right now VSCode is just not usable: git doesn't work nor the terminal and most of the extensions like C/C++, CMake Tools, etc. It's reverted to a simple text editor)

**Comment by tazerdev** [comment]
From the changelog:

> With these changes, the following are the distros known to work for the x64 desktop app:
> 
>     Ubuntu 16.04 and newer
>     Fedora 24 and newer
>     Debian 9 and newer
>     CentOS 8 and newer
> 
> Note: Our remote development components continue to use GLIBCXX 3.4.19, so there is no change in supported platforms.

So I guess my next question would be why, if it's no longer a supported platform, is CentOS7 being given an RPM that's meant for CentOS8?

**Comment by gjsjohnmurray** [comment]
> Is there any workaround known at this point ?

@dcourtois previous versions can be downloaded by going to https://code.visualstudio.com/updates and choosing the version from the sidebar, then using the download link on its release notes.

I assume you are using CentOS 7 as your desktop, right? And can't use any of the workarounds offered in the 1.53 release notes?
> A workaround for the other distros would be to install gcc-5 or higher toolchain to avoid the GLIBCXX error with native modules, but there is no guarantee that all components of the runtime will work fine. There is also the option of using our remote development suite to work with the older distros.

**Comment by RudiFueller** [comment]
If you are affected by this issue, let me tell you what I did:

- Download a recent gcc-tarball from a GNU mirror + compile
- Take the content of x86_64-pc-linux-gnu/libstdc++-v3/src/.libs and copy to a dir of your choice (eg. ~/lib64/libstdc++-v3)
- Write a start wrapper for vscode, put your extended LD_LIBRARY_PATH in the wrapper
- Open vscode with the wrapper -> done.

**Comment by Tyriar** [comment]
Sorry but we needed to move forward to support the new version of Electron. Here are your options:

- Upgrade your OS
- If you're remoting into the machine you can use the remote development extensions or codespaces because this only affects Electron (desktop), not node (remote server)
- Try install the compatible binaries as suggested above https://github.com/microsoft/vscode/issues/115784#issuecomment-774087742, YMMV with this approach
- Use an older version of VS Code - not only will you miss out on new updates but also security updates so this isn't recommended

**Comment by tazerdev** [comment]
@Tyriar 

Understood but the 1.53 RPM shouldn't install on a RHEL7/CentOS7 based system in the first place because RHEL7 has a standardized GLIBC that will be maintained throughout its lifecycle. The rpmspec should include a requires statement based on GLIBC version and the rpms should be provided in an EL compatible repository (i.e., RHEL7 and RHEL8 rpms are maintained in separate locations.)

As it stands VS Code is breaking the packaging standards of the platform and causing problems for users. If you can direct me to the appropriate area I'll gladly open a bug request there.

**Comment by Tyriar** [comment]
@tazerdev good point, @deepak1556 we need to update the deps in https://github.com/microsoft/vscode/blob/master/resources/linux/rpm/dependencies.json. Historically I just pull in whatever Chromium targets in their spec.

**Comment by hardik-singhal** [comment]
Unable to open vscode debugg console and terminal after vs code update. Is there any solution yet.
**I am using centos 7** and can't install any library mention in the error**

**Comment by elulcao** [comment]
Not only Centos has been affected, but also other flavors like OL7 which is a variant of the RH7, etc...
Some env cannot be upgraded/updated for different reasons, so I guess VSCode needs to evaluate how to
make compatible again this new Drop of code1.53
Sharing the stack trace for tracking purpose.

```bash
cat /etc/*release
Oracle Linux Server release 7.9
NAME="Oracle Linux Server"
VERSION="7.9"
ID="ol"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.9"
PRETTY_NAME="Oracle Linux Server 7.9"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:oracle:linux:7:9:server"
HOME_URL="https://linux.oracle.com/"
BUG_REPORT_URL="https://bugzilla.oracle.com/"

ORACLE_BUGZILLA_PRODUCT="Oracle Linux 7"
ORACLE_BUGZILLA_PRODUCT_VERSION=7.9
ORACLE_SUPPORT_PRODUCT="Oracle Linux"
ORACLE_SUPPORT_PRODUCT_VERSION=7.9
Red Hat Enterprise Linux Server release 7.9 (Maipo)
Oracle Linux Server release 7.9
```

```bash
[16111:0206/123246.162480:INFO:CONSOLE(618)] "%c  ERR color: #f33 /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /usr/share/code/resources/app/node_modules.asar.unpacked/vscode-sqlite3/build/Release/sqlite.node): Error: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /usr/share/code/resources/app/node_modules.asar.unpacked/vscode-sqlite3/build/Release/sqlite.node)
    at process.func [as dlopen] (electron/js2c/asar_bundle.js:5:1812)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1250:18)
    at Object.func [as .node] (electron/js2c/asar_bundle.js:5:2039)
    at Module.load (internal/modules/cjs/loader.js:1039:32)
    at Module._load (internal/modules/cjs/loader.js:932:14)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12738)
    at Module.require (internal/modules/cjs/loader.js:1079:19)
    at v (/usr/share/code/resources/app/out/vs/loader.js:3:12287)
    at Object.<anonymous> (/usr/share/code/resources/app/node_modules.asar/vscode-sqlite3/lib/sqlite3.js:1:77)
    at Module.u._compile (/usr/share/code/resources/app/out/vs/loader.js:3:12841)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1039:32)
    at Module._load (internal/modules/cjs/loader.js:932:14)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12738)
    at Module.require (internal/modules/cjs/loader.js:1079:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at t (/usr/share/code/resources/app/out/vs/loader.js:4:101)
    at r.load (/usr/share/code/resources/app/out/vs/loader.js:3:13249)
    at r.load (/usr/share/code/resources/app/out/vs/loader.js:3:10262)
    at c (/usr/share/code/resources/app/out/vs/loader.js:4:10314)
    at Object.errorback (/usr/share/code/resources/app/out/vs/loader.js:4:10435)
    at r.triggerErrorback (/usr/share/code/resources/app/out/vs/loader.js:3:10626)
    at /usr/share/code/resources/app/out/vs/loader.js:3:10332
    at r.load (/usr/share/code/resources/app/out/vs/loader.js:3:13266)
    at r.load (/usr/share/code/resources/app/out/vs/loader.js:3:10262)
    at c (/usr/share/code/resources/app/out/vs/loader.js:4:10314)
    at r._loadModule (/usr/share/code/resources/app/out/vs/loader.js:4:10444)
    at r._resolve (/usr/share/code/resources/app/out/vs/loader.js:5:452)
    at r.defineModule (/usr/share/code/resources/app/out/vs/loader.js:4:6145)
    at r._relativeRequire (/usr/share/code/resources/app/out/vs/loader.js:4:6831)
    at n (/usr/share/code/resources/app/out/vs/loader.js:4:9420)
    at /usr/share/code/resources/app/out/vs/code/electron-main/main.js:21:31382
    at new Promise (<anonymous>)
    at /usr/share/code/resources/app/out/vs/code/electron-main/main.js:21:31362
    at new Promise (<anonymous>)
    at n.doConnect (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:21:31342)
    at n.connect (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:21:30812)
    at new n (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:21:28194)
    at Ot.doInitialize (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:29:16863)
    at Ot.initialize (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:29:16753)
    at M.init (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:47:2577)
    at new M (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:47:2523)
    at lt.openFirstWindow (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:55:82995)
    at /usr/share/code/resources/app/out/vs/code/electron-main/main.js:55:77462
    at E.invokeFunction (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:26:25463)
    at lt.startup (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:55:77438)
    at async Y.startup (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:57:1448)", source: file:///usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (618)
```

**Comment by markusdd** [comment]
In my view this change is not acceptable. Especially  RHEL7 and derivatives are very widely used in enterprise environments and there is no way just 'to upgrade' as usually other commercial tools (in our case EDA simulators) do not even exist yet for RHEL8.

This drops a bomb into the workflow of many I would suggest.

At least there should be a very clear documentation on how to work around this and this should also be tested continuously for new releases in my view.

The recommended remote server workflows are not compatbile with all IT setups where VS Code is being used.

**Comment by dcourtois** [comment]
Hi, I just wanted to let other users of CentOS/RHEL7 that the method suggested by @RudiFueller works fine (at least for him and me) As for me, I couldn't find prebuilt packages of recent Gcc so I built one from scratch:

1. Go to https://gcc.gnu.org/mirrors.html, choose a mirror, and download a recent Gcc (I used the 5.5.0)
2. Unpack and build using something like:
```
installdir=/path/to/some/directory
./configure ./configure --prefix=$installdir CFLAGS="-O3 -g0" CXXFLAGS="-O3 -g0" LDFLAGS="-s" --disable-multilib
make -j10
make install
```
3. Create a `code.sh` script or something, set the executable flag (`chmod +x code.sh`) and put this inside:
```
export LD_LIBRARY_PATH=/path/to/some/directory/lib64:$LD_LIBRARY_PATH
/usr/bin/code
```

You will probably need to install a few dev libraries to build Gcc. Eventhough I'm a developer, my CentOS was still missing a few:
`sudo yum install gmp-devel mpfr-devel libmpc-devel`

VSCode seems to work fine like this.

**Comment by markusdd** [comment]
So I can also confirm we went a similar route and deployed a newer gcc version in a seperate path via our internal rpm repo and came up with a similar wrapper script.

I nevertheless would suggest that this should also be part of the test suite for VS Code, meaning:

1) Provide this workarounf in the docs
2) Have such an environment run in the regression for vscode to ensure stuff does not break over and over again
3) make a recommendation which gcc/addtll. libs to use.

Maybe there is also a way to provide a script repo or something that automates some of this stuff for users which are not as confident in compiling and deploying a whole new gcc by themselves.

**Comment by gjsjohnmurray** [comment]
In #116126 a RHEL7 user is reporting that the 'Report Issue' option from the Help menu broke with 1.53. Any RHEL7/CentOS7 users here willing to check that? Whether or not it's broken for you, please also say whether or not you've done the above workaround already.

**Comment by dcourtois** [comment]
@gjsjohnmurray I've tested the `Report Issue` on CentOS 7 and: works fine with the workaround, doesn't work at all without the workaround.

**Comment by deepak1556** [comment]
First of all sorry about this drop in support! There are two parts to the problem with support for RHEL7, CentOS7

1) They have not been officially listed as supported platforms by electron for a long time now https://www.electronjs.org/docs/tutorial/support#supported-platforms . What this means is that, we are on our known if any issues related to the runtime were to be found.

2) Update to electron 11 required us to update our build image which caused the regression to use newer `CXXABI` and the above workaround to use newer gcc toolchain should resolve it as also mentioned in the release notes https://code.visualstudio.com/updates/v1_53#_electron-11-update and this should resolve the issues related to dialog boxes not popping up or high cpu usage.

Even after fixing the above error there were reports of rendering issues which was discovered in our insiders testing https://github.com/electron/electron/issues/27480 , the issue was side-effect of refactor from chromium to use xcb instead of xlib for the display server communication. We currently don't have the scope to investigate this issue further.

With that said, we are happy to accept any PR that adds the workaround for fixing the `CXXABI` error to the docs,  but we can't list the above versions of distro as officially supported by vscode.

**Comment by andrecp** [comment]
I understand this challenges but just want to flag that this caused massive pain to a lot of users that are in an older version of CentOS (usually enterprise...), it seems reasonable to me as a workaround to not have the new RPM be installed by default on CentOS-7 since it doesn't work at all. And then in the docs have the workaround if you want to install a more up-to-date version by building your own GCC and pointing to it.

**Comment by omenos** [comment]
I've been able to successfully build a functioning OSS version of VSCode on CentOS 7. Steps below:

```bash
# Prep work (as root or with sudo)
> yum -y install centos-release-scl epel-release
> yum -y install rh-nodejs10 devtoolset-9 make libX11-devel libxkbfile-devel libsecret-devel fakeroot git
> scl enable rh-nodejs10 "npm install --global yarn"

# Build
> scl enable rh-nodejs10 devtoolset-9 bash
> git clone -b 1.53.0 --depth 1 https://github.com/microsoft/vscode.git
> cd vscode
> yarn --frozen-lockfile --network-timeout 180000
# Potentially optional
> yarn npm-run-all --max_old_space_size=4095 -lp compile "electron x64" playwright-install download-builtin-extensions
# End optional
> yarn run gulp vscode-linux-x64-min
> yarn run gulp vscode-linux-x64-build-rpm
```

Your RPM will now be available in `.build/linux/rpm/x86_64/code-oss-<VERSION>.rpm`. You can either extract the contents with `rpm2cpio <RPM> | cpio -idm` and do what you want (like a central install), or install it directly.

A note, however: The Code OSS build does not support the Extensions Marketplace out of the box. You will need to add the following blob to the `resources/app/product.json` to enable it (extensions can still be installed through VSIX directly without it, though):

```json
	"extensionsGallery": {
		"serviceUrl": "https://marketplace.visualstudio.com/_apis/public/gallery",
		"cacheUrl": "https://vscode.blob.core.windows.net/gallery/index",
		"itemUrl": "https://marketplace.visualstudio.com/items",
		"controlUrl": "https://az764295.vo.msecnd.net/extensions/marketplace.json",
		"recommendationsUrl": "https://az764295.vo.msecnd.net/extensions/workspaceRecommendations.json.gz"
	},
```

I don't know if the proposed change to the RPM SPEC file will negatively impact this process in the future. The devtoolset here will statically link in the symbols needed directly into the code binary, which would normally be available at runtime in `libstdc++` from a GCC 5+ package. In this case I'm using GCC 9 as it's the latest DTS available.

**Comment by deepak1556** [comment]
> it seems reasonable to me as a workaround to not have the new RPM be installed by default on CentOS-7 since it doesn't work at all. And then in the docs have the workaround if you want to install a more up-to-date version by building your own GCC and pointing to it.

@andrecp yup that is the next set of actions to be done here.

**Comment by rkd628** [comment]
Ubuntu 14.04 also has that issue..

**Comment by IgorNovozhilov** [comment]
There is still a way to solve the problem with dependencies on older OS.
You can use an alternative package manager [snapcraft.io](https://snapcraft.io/). ([installing-snap-on-centos](https://snapcraft.io/docs/installing-snap-on-centos))
In the sandbox, everything works fine, depending on the operating system.
I'm use CentOS-7 and VSCode in snap, and everything works fine for me.
Use it: https://snapcraft.io/code

> PS: Or use a different package manager, for example [Flatpak](https://flatpak.org/setup/CentOS/): [app Visual Studio Code](https://flathub.org/apps/details/com.visualstudio.code)

**Comment by sreerajkksd** [comment]
Compiling gcc along with the libraries and setting LD_LIBRARY_PATH solution worked for me.

This is the detailed set of steps:

- Install the required libraries for building the packages
```
yum install -y gmp-devel mpfr-devel libmpc-devel
```

- Get the gcc archives
```
curl https://ftp.gnu.org/gnu/gcc/gcc-9.3.0/gcc-9.3.0.tar.gz -O
tar xvf gcc-9.3.0.tar.gz
```
 - Build the package
```
install_dir=/directory/to/install/
mkdir gcc-9.3.0-build
cd gcc-9.3.0-build
../gcc-9.3.0/configure --prefix=$install_dir  --enable-languages=c,c++ --disable-multilib
make -j$(nproc) && make install
```
- Create a wrapper script for running vscode
```
#!/usr/bin/bash
exec env LD_LIBRARY_PATH=/directory/to/install/lib64:$LD_LIBRARY_PATH /usr/bin/code
```

**Comment by Luisss** [comment]
While waiting for a fix, a downgrade makes vscode functional again.
`yum downgrade code.x86_64`
In my case, I had to downgrade two times.

**Comment by eustaqxchange** [comment]
> Compiling gcc along with the libraries and setting LD_LIBRARY_PATH solution worked for me.
> 
> This is the detailed set of steps:
> 
> * Install the required libraries for building the packages
> 
> ```
> yum install -y gmp-devel mpfr-devel libmpc-devel
> ```
> 
> * Get the gcc archives
> 
> ```
> curl https://ftp.gnu.org/gnu/gcc/gcc-9.3.0/gcc-9.3.0.tar.gz -O
> tar xvf gcc-9.3.0.tar.gz
> ```
> 
> * Build the package
> 
> ```
> install_dir=/directory/to/install/
> mkdir gcc-9.3.0-build
> cd gcc-9.3.0-build
> ../gcc-9.3.0/configure --prefix=$install_dir  --enable-languages=c,c++ --disable-multilib
> make -j$(nproc) && make install
> ```
> 
> * Create a wrapper script for running vscode
> 
> ```
> #!/usr/bin/bash
> exec env LD_LIBRARY_PATH=/directory/to/install/lib64:$LD_LIBRARY_PATH /usr/bin/code
> ```

Can you elaborate the steps 3 and 4 should I copy paste this to notepad? Maybe upload a video tutorial. Thanks.

**Comment by denizobi** [comment]
The terminal process failed to launch: A native exception occurred during launch (/lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /usr/share/code/resources/app/node_modules.asar.unpacked/node-pty/build/Release/pty.node)).

**Comment by mintyc** [comment]
Centos 7 workaround assuming vscode repo is available
sudo yum remove code
sudo yum install code-1.51.1
(and set updates to None in vscode)

Note this picks up (and rolls back) any extensions already set in the newer version sominimal reconfiguration required

**Comment by israelKusayev** [comment]
This answer should help! (I had the same issue on centOS 7, but now I'm able to run the latest version of vscode without any problems)
https://github.com/cdr/code-server/issues/347#issuecomment-482670081

**Comment by andre-rosa** [comment]
An [AppImage](https://appimage.org/) should be provided.

**Comment by sreerajkksd** [comment]
> 
> 
> > Compiling gcc along with the libraries and setting LD_LIBRARY_PATH solution worked for me.
> > This is the detailed set of steps:
> > 
> > * Install the required libraries for building the packages
> > 
> > ```
> > yum install -y gmp-devel mpfr-devel libmpc-devel
> > ```
> > 
> > 
> > 
> > * Get the gcc archives
> > 
> > ```
> > curl https://ftp.gnu.org/gnu/gcc/gcc-9.3.0/gcc-9.3.0.tar.gz -O
> > tar xvf gcc-9.3.0.tar.gz
> > ```
> > 
> > 
> > 
> > * Build the package
> > 
> > ```
> > install_dir=/directory/to/install/
> > mkdir gcc-9.3.0-build
> > cd gcc-9.3.0-build
> > ../gcc-9.3.0/configure --prefix=$install_dir  --enable-languages=c,c++ --disable-multilib
> > make -j$(nproc) && make install
> > ```
> > 
> > 
> > 
> > * Create a wrapper script for running vscode
> > 
> > ```
> > #!/usr/bin/bash
> > exec env LD_LIBRARY_PATH=/directory/to/install/lib64:$LD_LIBRARY_PATH /usr/bin/code
> > ```
> 
> Can you elaborate the steps 3 and 4 should I copy paste this to notepad? Maybe upload a video tutorial. Thanks.

I self documented the steps 3 and 4.

**Comment by robdmorton** [comment]
I can't change my gcc version due to project constraints.   

Is there a solution coming in which centos7 users don't have to change their environment to accommodate VSCode?

**Comment by jfgudk** [comment]
Running downgrade 3 times resolved it for me.
```
[~]$ sudo yum downgrade code.x86_64
...
Resolving Dependencies
--> Running transaction check
---> Package code.x86_64 0:1.52.1-1608137084.el7 will be a downgrade
---> Package code.x86_64 0:1.53.0-1612368438.el7 will be erased
--> Finished Dependency Resolution

Dependencies Resolved
...

Removed:
  code.x86_64 0:1.53.0-1612368438.el7                                                                                                                                                                              

Installed:
  code.x86_64 0:1.52.1-1608137084.el7                                                                                                                                                                              

Complete!```

**Comment by cameronkerrnz** [comment]
I was planning to rebuild my development VM, but can't afford to do that mid-sprint. The following will downgrade in a single-step to the previous version (that I was using).

```
sudo yum downgrade code-1.52.1-1608137084.el7.x86_64
```

I'm on CentOS 7 because I do a lot of Ansible development currently, and some of those actions are local actions.

A potentially better way for VSCode users to avoid introducing binding constraints between project environments and workstation environments would be to use VSCode Dev Containers... particularly great if you're doing anything with the likes of JRuby (recent headache) but even then that can end up with some leaky abstractions (eg. If you're Docker Desktop environment is using WSL2, last time I looked (which _was_ prior to last significant update of WSL2), Ansible was behaving inconsistently.

**Comment by GitMensch** [comment]
As @Tyriar has said that the intermediate fix of not bringing breaking updates I've created a PR to do this (it actually just changes one byte...).

I definitely recommend to NOT replace libstdc++ to an unsupported version in the system path (suggestion at linked cdr/code-server issue) as this potentially breaks the complete system.
I can recommend having a single system library in a different place (self-build or possibly even from an external app that ships it) pre-loaded with `LD_PRE_LOAD` in a wrapper (I had to do something similar to get code running via X11 forwarding) - that only effects one single application.

I also agree that an AppImage would be a nice thing to publish, but that was declined for vscode in https://github.com/microsoft/vscode/issues/10857#issuecomment-552500236.

Question concerning the "mid-term-solution":
Is it considered/planned to package or statically link libcstd++?

**Comment by markusdd** [comment]
I think it should be no problem to offer a seperate 'compat' rpm/deb package that can be installed alongside that puts a fitting libstdc++ into the right path and carries the main package as a dependency and vice versa.

This would solve all the headaches and saves users from having to tinker themselves.

**Comment by GitMensch** [comment]
Nice idea but this won't work because the compat EL7 rpm cannot depend on the soon EL8 main rpm (and it would have to overwrite the final binaries).
@markusdd Despite of this: can you check for a GLIBXX-only rpm (which does NOT overwrite the system libstdc++)? If that's available then people can `yum install` it and only need the LD_PRELOAD,

**Comment by markusdd** [comment]
why not setup a seperate EL7 repo?

As it's a very, and I mean VERY, popular Distribution I do not see why this wouldn't make sense. Builds are automated, so where's the problem?

**Comment by s2terminal** [comment]
When I wanted to run `$ yum update` before this issue was resolved, it worked to put the following in `/etc/yum.conf`

```
exclude=code
```

It will no longer update the VS Code when `$ yum update` is performed.

**Comment by tazerdev** [comment]
@s2terminal yum versionlock is a bit easier to work with and recommended by Red Hat for pinning versions. The better way is to disable or remove the yum repository from your system entirely because, as stated earlier in this thread, the el7 platform has been abandoned so there's no point in even looking for updates.

You'll then need to set the update policy for code to none in your settings:

```json
"update.mode": "none"
```

For giggles I did update to the version that was released this morning and it doesn't work at all. In fact it froze with 100% cpu usage and had to be kill -9'd. While this isn't technically a bug, the packages are still being falsely marked as el7 which indicates compatibility with RHEL7.X. For instance:

```
code-1.54.2-1615424933.el7.x86_64.rpm
```

This would be akin to releasing 64 bit binaries but marking the packages i386. Again, I'm not requesting developers maintain compatibility but name the RPMs properly, and provide appropriate repositories, or people will rightly come away from a VS Code on RHEL7 install thinking this project is a disaster, which will negatively affect Microsoft's reputation.

**Comment by GitMensch** [comment]
And #117994 should fix the wrong claim... (maybe should include an explicit CXXABI, but I can't test for the version define so will leave that to others)

**Comment by LucienMP** [comment]
Also the docs page needs to be more extensive/updated to reflect EOL for CentOS7/RHEL7.
( https://code.visualstudio.com/docs/supporting/requirements )

Alternatively maybe it should reflect supported OS vs VSCode version so people can make the decision to use, and less confusion post install (as the install silently fails)

**Comment by GitMensch** [comment]
That's definitely a good point, it still lists RHEL7, but that actually don't work.

**Comment by appurist** [comment]
I am the technical team leader in an enterprise (government) environment where my development laptop is spec'd to match the target environment, which uses CentOS 7 for LTS stability (support extends to 2024, well beyond CentOS 8 which ends this year). My software will be installed on server boxes on Navy ships, and I need to develop on a Navy-supplied laptop running the same environment.

This sudden -- and as far as I can see unannounced -- nuclear bomb in my development toolchain is going to cost us tremendously, unless it can be rolled back quickly.

Somewhere between a 1.52 and 1.54 update, our primary development tooling was completely broken, without even an error message being displayed. I've already lost a lot of time [reporting and diagnosing](https://github.com/microsoft/vscode/issues/118987) a sudden complete hang of VS Code after an upgraded, and don't really have good alternatives here now that I know what the problem is.

This sudden change to drop support for major LTS OS releases is highly unprofessional, and a shock on an otherwise wonderful history, with VS code setting an example for how development tools should be done. Until now. It is extremely disruptive to the community, and I hope the VS Code team will rethink it.

Management above me will see this as a clear sign that Microsoft should not be trusted with Linux development tools, and their commitment to support Linux. I suspect the VS Code team has no plans to drop support for Windows 7, 8, or 8.1, so dropoing support for CentOS 7 and RHEL 7 stands out as ... inappropriate. I assume there is a cost-benefit trade-off forcing this change, but I want to make it clear that, given a working 1.52 on these operating systems, I cannot imagine a benefit for 1.54 that justifies this cost.

I can temporarily workaround this problem by pinning an old version, using one of the techniques mentioned above, but blocking all VS Code updates is probably not a smart or sustainable approach, especially in a government military environment where security is such an overwhelming priority.

I will need to report to them that I must now look at alternative tooling for our projects. I am not really looking forward to switching to WebStorm or ATOM or one of the alternatives. I've grown fond of Code and this is more than disappointing, and this will be seen by many as an "I told you so" moment, a betrayal of Linux developers by Microsoft. I know (or at least think) this isn't accurate, however I'm not going to be in a position to defend this decision.

**Comment by sifordtj** [comment]
> I am the technical team leader in an enterprise (government) environment where my development laptop is spec'd to match the target environment, which uses CentOS 7 for LTS stability (support extends to 2024, well beyond CentOS 8 which ends this year). My software will be installed on server boxes on Navy ships, and I need to develop on a Navy-supplied laptop running the same environment.
> 
> This sudden -- and as far as I can see unannounced -- nuclear bomb in my development toolchain is going to cost us tremendously, unless it can be rolled back quickly.
> 
> Somewhere between a 1.52 and 1.54 update, our primary development tooling was completely broken, without even an error message being displayed. I've already lost a lot of time [reporting and diagnosing](https://github.com/microsoft/vscode/issues/118987) a sudden complete hang of VS Code after an upgraded, and don't really have good alternatives here now that I know what the problem is.
> 
> This sudden change to drop support for major LTS OS releases is highly unprofessional, and a shock on an otherwise wonderful history, with VS code setting an example for how development tools should be done. Until now. It is extremely disruptive to the community, and I hope the VS Code team will rethink it.
> 
> Management above me will see this as a clear sign that Microsoft should not be trusted with Linux development tools, and their commitment to support Linux. I suspect the VS Code team has no plans to drop support for Windows 7, 8, or 8.1, so dropoing support for CentOS 7 and RHEL 7 stands out as ... inappropriate. I assume there is a cost-benefit trade-off forcing this change, but I want to make it clear that, given a working 1.52 on these operating systems, I cannot imagine a benefit for 1.54 that justifies this cost.
> 
> I can temporarily workaround this problem by pinning an old version, using one of the techniques mentioned above, but blocking all VS Code updates is probably not a smart or sustainable approach, especially in a government military environment where security is such an overwhelming priority.
> 
> I will need to report to them that I must now look at alternative tooling for our projects. I am not really looking forward to switching to WebStorm or ATOM or one of the alternatives. I've grown fond of Code and this is more than disappointing, and this will be seen by many as an "I told you so" moment, a betrayal of Linux developers by Microsoft. I know (or at least think) this isn't accurate, however I'm not going to be in a position to defend this decision.

I couldn't agree more (and said as much above).  Once more and more people find this post, we will see more and more complaints.  Like you, I'm doing government work.  We're clearly not going to "upgrade" to CentOS 8 when it reaches its EOL in months compared to YEARS for CentOS 7.  This is an absolutely huge blunder that needs to be rectified.  You can't just remove CentOS 7 support when there is literally no other CentOS that has Long Term Support.  It makes ZERO sense.  None.  And then to do it with just the tiniest footnote in the January release notes is unacceptable ... all while keeping the software available for auto-update (even though clearly the VSCode team already knows that they removed the ability for it to work on CentOS 7).

**Comment by markusdd** [comment]
I also do not understand why there isn'T a pragmatic solution.

Make a EL7-specific repo, package a fitting libstc++ alongside vscode as dependency, done.

No hacked start-scripts, no wild out-of-path installations of compiler and libraries, no modified LD_LIBRARY_PATH.

I can understand why the developers had to make the switch, but I have to agree, abondoning the most widely used enterprise distirbution is very, very short-sighted. (especially without warning and announcement)

**Comment by appurist** [comment]
The [January release notes](https://code.visualstudio.com/updates/v1_53#_electron-11-update) claim:

> ### Electron 11 update
> In this milestone, we finished the exploration to bundle Electron 11 into VS Code, thanks to everyone involved with testing and self-hosting on insiders. This is a major Electron release and comes with Chromium 87.0.4280.141 and Node.js 12.18.3.
> 
> #### Breaking change
> 
> As a side effect of this update, we had to bump our build image to use Ubuntu-18.04 for x64 linux machines to consume Electron. This update raised the minimum GLIBCXX requirement to 3.4.22 for our native modules, which breaks support for older distros on desktop.

But my CentOS 7 app is an Electron 11.3 app. Builds and runs fine on CentOS 7 without any special steps.  Not sure what "we had to" means here. Perhaps this is all a misunderstanding

**Comment by appurist** [comment]
Specific steps for anyone else finding this problem and needing to pin VS Code to 1.52:

`sudo yum downgrade code-1.52.1-1608137084.el7.x86_6`
or if you don't have it installed:
`sudo yum install code-1.52.1-1608137084.el7.x86_6`

`sudo yum install yum-plugin-versionlock`
(if it's not already installed)
`sudo yum versionlock code`
(to lock it at 1.52)

Note that it's still possible to test more recent versions alongside 1.52 by installing the [nightly "insiders" build](https://code.visualstudio.com/insiders/), which installs a separate `code-insiders` command.

**Comment by LucienMP** [comment]
You can follow the steps on this stackoverflow.com to build GCC 10.0.1 (no need to install) and use LD_LIBRARY_PATH/LD_PRELOAD to let vscode run in CentOS7 (7.8.2003 as per below article and on mine 7.9.2009)

https://stackoverflow.com/a/37355018/1120779

```
# required libraries: (some may already have been installed)
sudo yum install libmpc-devel mpfr-devel gmp-devel

# if dnf install libmpc-devel is not working try:
sudo yum install libmpc-devel

# install zlib
sudo yum install zlib-devel*

# Get Package
wget https://ftp.gnu.org/gnu/gcc/gcc-10.1.0/gcc-10.1.0.tar.gz

# Expand and Compile, expect about 6GB of used space
tar -zcf gcc-10.1.0.tar.gz
cd gcc-10.1.0
mkdir build
cd build
../configure --with-system-zlib --disable-multilib --enable-languages=c,c++

# this may take around an hour or more to finish (depending on your cpu speed)
make -j 8 

LD_PRELOAD=$(readlink -m ./x86_64-pc-linux-gnu/libstdc++-v3/src/.libs/libstdc++.so.6.0.28)
code
```

**Comment by GitMensch** [comment]
> You can follow the steps on this stackoverflow.com to build GCC 10.0.1

thanks but those were posted multiple times already - this is quite a big hack (it is even out of scope for "workaround") and if you cannot use `yum install` because you cannot install anything then you'd need to also build all the dependencies on your own before. Multiple hours work(aroundhack) and build time just to create a glibc library that works around a bug in vscode publishing)...

Don't get me wrong, I'd totally like CentOS7 to be supported (in this case that likely would happen with a different rpm repo where vscode+included glibc is found and the main repo version having the broken dependency claim fixed [which #117994 is about]) - but without a commitment of the vscode team I don't see that coming...

**Comment by markusdd** [comment]
The whole point is: Most people dicussing here are probably insiders in a sense that they have found this workaround already or they maintain some sort of devlopment infrastructure for multiple people. Such a hack is nice locally, but when you have whole teams working on their machines and multiple servers, such stuff gets messy real quick.

So I ask again: Is there a way for an official EL7 repo which has a dependency packages with a known good, CI-tested libstd++?
Is the VS code maintainer team willing to pursue that?

**Comment by LucienMP** [comment]
> thanks but those were posted multiple times already ... 
I am leaving this ugly hack here for the years ahead and because nobody listed the exact steps needed since there will likely be those that follow without the requisite skills or in future the version that will work with least hassle.

Like @appurist above my group rolls out CentOS7 for our device support, it costs millions for the device and so LTS is important, and we dont easily upgrade our customers, or ourselves.

> Don't get me wrong, I'd totally like CentOS7 to be supported
Same here, this is one of the worst issues with Linux distro fragmentation, so I guess were stuck with 1.52

**Comment by GitMensch** [comment]
> this is one of the worst issues with Linux distro fragmentation

I don't think it is reasonable to complain about GNU/Linux distros because Microsoft commits something with a broken rpm spec to CentOS repos (explicit telling "I'm running fine with EL7 and I don't have s special CXXABI requirement either" when it does not).

This part is a packaging issue, which is a question about "does the packager cares".
And if the packager cares it is also the question of the maintainers to decide if they want to support an nearly 4+ years left LTS system (knowing that they likely get a bunch of issues about "xyz does not work on CentOS7" [see closed issues about some, many of those point here] which then need to be fixed) or not. If the answer to the second one is "no" (which is a strategic decision by the repo owner which is named "microsoft" as would be a "yes") and the package cares, then the only thing he can to is applying changes like #117994 and republish the broken 1.53 versions.

**Comment by qwertycody** [comment]
+1 - Impacting many managed development environments internally

**Comment by appurist** [comment]
Thanks @LucienMP for posting the custom build steps for the dependency upgrades, which are shorter than what I've seen elsewhere.  I'm running through them as a test on my development machine now (build does take a long time) but it has all worked so far (except there's a typo on the unzip, where `-zcf` should read `-zxf`). I think it is good to document clearly any steps that may be workarounds at some installations.

This won't be appropriate for my coworkers, but it good in general to know how to upgrade these dependencies, so I'm giving it a spin.

VS Code devs:
- Consider rolling back the change in 1.53, since the reason provided (to build an Electron 11 version of Code) is not actually required to do so.
- Also just because the negative impact on enterprise developers is almost overwhelming, and switching to supporting a short-lived distro (CentOS 8) with 9 months of life left, over a LTS enterprise distro (CentOS 7) makes little sense and may mean we're going to need to go through this again in less than a year.
- It violates the good engineering practice of building on your primary target system, to maximize effective testing and early detection of problems. The custom build + LD_ variable kludge significantly alters the system from the target production environment, masking the early discovery of potential problems.

Me:
- I'll speak to stakeholders about choosing an alternate LTS distro for production. I don't think they'll go for this, as it's a government/military environment that is standardized wider than my project, and quite slow to change. Clearly CentOS 8 is not appropriate either, given its short support life, but perhaps I could encourage RHEL8 or some other LTS distro be selected.

**Comment by appurist** [comment]
FYI, the instructions for running a newer code with [the steps from @LucienMP](https://github.com/microsoft/vscode/issues/115784#issuecomment-800541823)  **do** seem to work successfully for me. I was able to run a recent 1.55-insiders version with the LD_PRELOAD variable set. It's an ugly hack that doesn't allow you to just run code normally, takes over an hour to build, and I don't think I can suggest it to coworkers, but it _does_ allow upgrades to Code on any machine where it's done, and gets a single developer past a show-stopper problem.

**Comment by GitMensch** [comment]
If you want to you can just copy the generated libraries to another system which is similar enough (as long as you don't change the architecture, but likely most of these centos/RHEL7 will be AMD64 in any case). This removes the need to build it over and over again (and you can also find out if a single shared object is enough to copy).
... and that also would be the option outlined above to package code along this library in a separate repo (or even always, works as long as -R can be used during linking, which likely is the case for RPM builds).

**Comment by sifordtj** [comment]
> FYI, the instructions for running a newer code with [the steps from @LucienMP](https://github.com/microsoft/vscode/issues/115784#issuecomment-800541823) **do** seem to work successfully for me. I was able to run a recent 1.55-insiders version with the LD_PRELOAD variable set. It's an ugly hack that doesn't allow you to just run code normally, takes over an hour to build, and I don't think I can suggest it to coworkers, but it _does_ allow upgrades to Code on any machine where it's done, and gets a single developer past a show-stopper problem.

For you, yes.  For others like me, no.  I meticulously followed the steps and its a no-go.  It produces the Makefile just fine.  But when actually building, it doesn't succeed.  I tried with different versions (fresh each time) too - no luck.  I'm not a linux expert by any means but I know my way around pretty good.

**Comment by LucienMP** [comment]
> ... no-go ... 
You maybe missing other packages, quite likely - always a nightmare with these things.
What error did you get, if you want we can try to get something more robust?

You can get some more information as below, and also try putting your whole compile output into a log?
```
[lmurray]$ lsb_release -a
LSB Version:	:core-4.1-amd64:core-4.1-noarch:cxx-4.1-amd64:cxx-4.1-noarch:desktop-4.1-amd64:desktop-4.1-noarch:languages-4.1-amd64:languages-4.1-noarch:printing-4.1-amd64:printing-4.1-noarch
Distributor ID:	CentOS
Description:	CentOS Linux release 7.9.2009 (Core)
Release:	7.9.2009
Codename:	Core
```

```
[lmurray]$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/4.8.5/lto-wrapper
Target: x86_64-redhat-linux
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-bootstrap --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-linker-hash-style=gnu --enable-languages=c,c++,objc,obj-c++,java,fortran,ada,go,lto --enable-plugin --enable-initfini-array --disable-libgcj --with-isl=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/isl-install --with-cloog=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/cloog-install --enable-gnu-indirect-function --with-tune=generic --with-arch_32=x86-64 --build=x86_64-redhat-linux
Thread model: posix
gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC) 
```

**Comment by nfstern** [comment]
> Specific steps for anyone else finding this problem and needing to pin VS Code to 1.52 in order to not have their toolchain completely disrupted:
> 
> `sudo yum downgrade code-1.52.1-1608137084.el7.x86_6`
> or if you don't have it installed:
> `sudo yum install code-1.52.1-1608137084.el7.x86_6`
> 
> `sudo yum install yum-plugin-versionlock`
> (if it's not already installed)
> `sudo yum versionlock code`
> (to lock it at 1.52)
> 
> Note that it's still possible to test more recent versions alongside 1.52 by installing the [nightly "insiders" build](https://code.visualstudio.com/insiders/), which installs a separate `code-insiders` command.
> 
> [Thanks to @tazerdev for mentioning versionlock above, and all the extremely helpful comments by him and others here.]

Thank you for posting this, it helped me work around the issue.

**Comment by tilleyc** [comment]
Just adding on here - it's completely unprofessional to release a breaking change for a major distro, acknowledge it, and then tell people to pound sand. Maybe that flies with Windows software, but with repositories being what they are for Linux, you absolutely need to own and actively manage your packaging.

Please either commit to an official fix, or remove all versions above 1.52 from el7 availability.

**Comment by tazerdev** [comment]
One side effect that limits the usefulness of staying on 1.52 is that as plugins are updated they increasingly become incompatible. For instance, Microsoft's Python/Jupyter plugins no longer work but instead fail with the below message:

```
Cannot activate the 'Python' extension because it depends on the
'Jupyter' extension, which is not loaded. Would you like to reload the
window to load the extension?
```

But the Jupyter extension is active and loaded, and when you reload the window the same error comes up along with:

```
Sorry, something went wrong activating IntelliCode support
for Python. Please check the "Python" and "VS IntelliCode"
output windows for details.
```

Whereas everything worked after I first opened this issue, extensions are slowly beginning to become incompatible. As it is, python development in VS Code is effectively dead on RHEL7.

It may be worth opening up another issue, but I feel strongly that VS Code shouldn't allow incompatible plugins to be installed.

Also, It should be stated clearly and openly that VS Code and its official Microsoft plugins aren't versioned and that the project will only support the _very_ latest version of the operating systems on which it claims to be compatible. Which means as soon as RHEL9 is released, you should plan for VS Code to breakdown and stop working shortly thereafter and with little notice.

And while I may not agree with that kind of development cycle (I certainly can't complain about free software and VS Code has been a wonderful piece of software for me for years), at least knowing up front that a development tool will only be functional for part of my operating system's life cycle will inform me enough to choose a different piece of software and to avoid recommending it to others.

**Comment by GitMensch** [comment]
> I feel it's no longer necessary without plugin compatibility enforcement

Note: You can use old versions (download via the ms market place or, if you have that enabled (which is the default) and internet access working from vscode, directly from the extensions view (under the gear wheel). If they "once worked" they will work in the future, too.

But definitely: vscode check a plugin for compatibility before installing it (which needs an attribute that extension authors do have to provide), maybe only the jupiter plugin is broken in this case?

**Comment by tazerdev** [comment]
Thanks for the information @GitMensch. That's definitely a potential work around to extend the life of 1.52.x under RHEL7.

On a whim I did work through the process of getting the latest VSC to run in a podman container using the official RHEL8 Universal Base Image (registry.access.redhat.com/ubi8/ubi). Interestingly it exhibited the bug you referenced above (unresponsive right after start up.)

Since the container has ~/.vscode and ~/.config/Code directories mapped to my existing directories I suspected there's something corrupt in there. When I start with those directories empty, that problem disappears. Not a deal breaker for me as I just needs to copy some settings over.

Installing extensions and customizing settings seems to work when starting fresh. The only weird issues I've seen with this setup is:

1. libX11-xcb wasn't installed as a dependency. It's possible this is missing in the RPM spec file for RHEL8 but hasn't come up yet because it's almost always installed on desktop systems.
2. VS Code freezes when I try to resize the terminal window.
3. The detailed extension documentation doesn't render after selecting an extension on the explorer bar.

None of those are critical to my workflow. If this remains stable this will be my preferred work around until I upgrade to rhel8 as I don't have to rebuild glibc/libstdc++ and maintain them. To avoid clutter in this thread, my notes on this process are located at:

https://github.com/tazerdev/vscode-podman

I'll just need to disable updates and rebuild the container whenever an update is released.

**Comment by dx-coding** [comment]
I've had luck getting 1.54.3 running on RHEL7 by downloading [libstdcxx-ng from anaconda](https://anaconda.org/conda-forge/libstdcxx-ng), and just copying the library (lib/* from the tar) to /usr/share/code/.

I make no guarantees that this doesn't break anything

**Comment by toddpfaff** [comment]
Here's a solution that worked for me ...

- install ghettoforge repo for CentOS 7
```
yum --nogpg install https://mirror.ghettoforge.org/distributions/gf/gf-release-latest.gf.el7.noarch.rpm
```
- install gcc10-libstdc++.x86_64
```
yum install gcc10-libstdc++.x86_64
```
- create link to new libstdc++ where VSCode will find it at runtime
```
 ln -s /opt/gcc-10.2.1/usr/lib64/libstdc++.so.6 /usr/share/code/
```
- run code
```
code
```

**Comment by outjoker** [comment]
@toddpfaff the steps that you mentioned for centos7, can you kindly let me know if they would be applicable for rhel 7.9 as well?

**Comment by gjsjohnmurray** [comment]
@deepak1556 is the upcoming 1.56 also going to install on RHEL 7 / CentOS 7 and then not work properly there? If so, I guess I'm not understanding something about what can or can't be done in the packaging to prevent this.

**Comment by sifordtj** [comment]
> Here's a solution that worked for me ...
> yum --nogpg install https://mirror.ghettoforge.org/distributions/gf/gf-release-latest.gf.el7.noarch.rpm
> yum install gcc10-libstdc++.x86_64
>  ln -s /opt/gcc-10.2.1/usr/lib64/libstdc++.so.6 /usr/share/code/
> code

Thank you toddpfaff!  This worked perfectly (except I had to add the requisite sudo in front).  Finally an easy solution to a horrible problem.

**Comment by GitMensch** [comment]
> is the upcoming 1.56 also going to install on RHEL 7 / CentOS 7 and then not work properly there? If so, I guess I'm not understanding something about what can or can't be done in the packaging to prevent this.

Hm, we changed the code.spec.template but obviously that template is later (possibly) used to create the final spec file which is then used to create the RPM - Where is that done (and does it actually use the current template)?

**Comment by J-M0** [comment]
@GitMensch, just changing the name of the RPM isn't enough. You could add `libstdc++.so.6(CXXABI_1.3.9)` to `resources/linux/rpm/dependencies.json` to make sure users on EL7 can't install VS Code, but that will make for an ugly error message when users try to update:

```
Resolving Dependencies
--> Running transaction check
---> Package code-1.56.0.x86_64 1620166346.el8 will be updated
---> Package code-1.56.0.x86_64 1620166346.el8 will be an update
--> Processing Dependency: libstdc++.so.6(CXXABI_1.3.9) for package: code-1.56.0-1620166346.el8.x86_64
--> Finished Dependency Resolution
Error: Package: code-1.56.0-1620166346.el8.x86_64
           Requires: libstdc++.so.6(CXXABI_1.3.9)
**********************************************************************
yum can be configured to try to resolve such errors by temporarily enabling
disabled repos and searching for missing dependencies.
To enable this functionality please set 'notify_only=0' in /etc/yum/pluginconf.d/search-disabled-repos.conf
**********************************************************************

Error: Package: code-1.56.0-1620166346.el8.x86_64
           Requires: libstdc++.so.6(CXXABI_1.3.9)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

The best thing to do would be to reorganize the physical layout of the repository at packages.microsoft.com. Take a look at how the Extra Packages for Enterprise Linux does it: https://mirror.us-midwest-1.nexcess.net/epel/

Each supported distro is in its own directory containing directories matching the supported arches. This allows them to use a simple `.repo` file for all supported platforms by leveraging variables provided by `yum` and `dnf`. Microsoft doesn't have to do exactly like EPEL does, but can do a simple layout like this would work:

```
packages.microsoft.com/yumrepos/vscode
|-- 7
|   |-- code-1.52.1-1608136323.el7.armv7hl.rpm
|   |-- code-1.52.1-1608136411.el7.aarch64.rpm
|   |-- code-1.52.1-1608137084.el7.x86_64.rpm
|   `-- repodata
|       |-- 0a69337359d057c65faaec062e594b62bb43932164b9bca4ecf96690c550f0fe-filelists.xml.gz
|       |-- 1f302bfb2001c1ef9d6c40e7c1059b8a73441a9ba2fae673136d107811695262-primary.sqlite.bz2
|       |-- 3e4e5817d5cfe92ce866fa07b7d6ed417d1f998182a4289fae3fd2e7025563aa-other.xml.gz
|       |-- 816219971e8ec457d8b4eb984795e9b1bd67d44dd6cea1c685a9e8b30eaeb26e-filelists.sqlite.bz2
|       |-- abe814ff6a989cf06f477a04ca15583340294f7124f3d15dd39cb8ba19b63a14-other.sqlite.bz2
|       |-- b602426435977020abda77e7289e12c7b8ef9a38e9b99ac934c595a105371fa8-primary.xml.gz
|       `-- repomd.xml
`-- 8
    |-- code-1.56.0-1620166175.el8.aarch64.rpm
    |-- code-1.56.0-1620166250.el8.armv7hl.rpm
    |-- code-1.56.0-1620166346.el8.x86_64.rpm
    `-- repodata
        |-- 453f7766cf3481baa5d5b81841dc083bf65645ce3eff15fafe6daabef0a63878-other.xml.gz
        |-- 46796e57cea3353b87dee2845ba6062b24049181887de04811f2ca599f33ae34-primary.sqlite.bz2
        |-- 6a7ea7e7f614d0e6ecb6cdc0faf3ca0bee8c17c24f1f9bf23803a7ebf11e60d8-filelists.xml.gz
        |-- 8df55284c130e7b0a7af16ddf6533ca41f7c65252fe9d6e0a3fbc889dc08272e-primary.xml.gz
        |-- 9c7e3782270a6b6cb6ebb08b02198c568d573500d436e7ee955bfbc354e3093a-other.sqlite.bz2
        |-- bf68f018a8949c4a8555d6efbab35458857addb38fb373bdde2a967556c7ca45-filelists.sqlite.bz2
        `-- repomd.xml
```

then MS needs to tell people what to put in `/etc/yum.repos.d/vscode.repo`, `7` for people on Enterprise Linux 7 and `8` for people on Enterprise Linux 8 and Fedora >= 22:

```
[code]
name=Visual Studio Code
baseurl=https://packages.microsoft.com/yumrepos/vscode/**<7 or 8>**
enabled=1
gpgcheck=1
gpgkey=https://packages.microsoft.com/keys/microsoft.asc
```

This allows for two good things:

1. If MS wanted to keep supporting VS Code on EL7, they could include a libstdc++ package in the `7` repo that would be used by VS Code to keep working
2. When this inevitably happens again, users will silently stop receiving updates rather than have broken software automatically installed on their machines.

-----
Finally, the best idea of all might actually be to use the existing `fedora`, `rhel`, and `centos` directories on packages.microsoft.com. Those are already set up more or less like standard multi-distro repos.

**Comment by fncambres** [comment]
> Aquí hay una solución que funcionó para mí ...
> 
> * instalar ghettoforge repo para CentOS 7
> 
> ```
> yum --nogpg install https://mirror.ghettoforge.org/distributions/gf/gf-release-latest.gf.el7.noarch.rpm
> ```
> 
> * instalar gcc10-libstdc ++. x86_64
> 
> ```
> yum install gcc10-libstdc++.x86_64
> ```
> 
> * crear un enlace a la nueva libstdc ++ donde VSCode lo encontrará en tiempo de ejecución
> 
> ```
>  ln -s /opt/gcc-10.2.1/usr/lib64/libstdc++.so.6 /usr/share/code/
> ```
> 
> * Ejecutar código
> 
> ```
> code
> ```

Gracias me funciono

**Comment by OlafRocket** [comment]
There are a couple of valid suggestions to keep the Centos 7 compatibility, is Microsoft going to comment on this?
Centos 7 is still supported at least until June 30, 2024 and is still widely used. for example in the VFX industry.

**Comment by SonGokussj4** [comment]
Hi. I'm IT at our company so on my DEV machine I've added `mirror.ghettoforge.org/...` and installed `gcc10-libstdc++` lib and it's **working just fine.**
But I can't do that for all of our users. (no approved installation of "unknown sources" which ghettoforge is)

We're using shared linux path where I've unpacked newest code and it broke everything for all users.
Any thought how to do this for all users in our network? Or we have to stay on the 1.53 version...?

Code is on shared drive: `/SHARED/apps/vscode...` (with `/SHARED/apps/bin` linking `../vscode/code`)
That is mounted `/SHARED -> /sharedapps/...`
Each user has extended PATH for `/sharedapps/bin`
Once a time I just refresh the vscode folder from newest tar.gz.

**Comment by GitMensch** [comment]
> Any thought how to do this for all users in our network?

Using the ghettoforge approach brings you a local libstdc++, instead of the link just copy it to the shared vscode folder, then run `ldd` on that to check if there are more libraries you need to copy.

> Or we have to stay on the 1.53 version...?

I'd personally likely stay with 1.53 until there's an official MS-Repo that solves this, [@J-M0 outlined the solution](https://github.com/microsoft/vscode/issues/115784#issuecomment-834835385)... or try to switch to Theia (at least check if it works with older libstdc++).

**Comment by tawani** [comment]
> Here's a solution that worked for me ...
> 
> * install ghettoforge repo for CentOS 7
> 
> ```
> yum --nogpg install https://mirror.ghettoforge.org/distributions/gf/gf-release-latest.gf.el7.noarch.rpm
> ```
> 
> * install gcc10-libstdc++.x86_64
> 
> ```
> yum install gcc10-libstdc++.x86_64
> ```
> 
> * create link to new libstdc++ where VSCode will find it at runtime
> 
> ```
>  ln -s /opt/gcc-10.2.1/usr/lib64/libstdc++.so.6 /usr/share/code/
> ```
> 
> * run code

Also worked on CentOS 8

**Comment by keegandent** [comment]
Was this not a consideration when the motion to upgrade to Electron 11 was being evaluated? RHEL 8 just turned two years old; RHEL 7 is hardly a legacy OS, and I'd struggle to think of any features compelling enough to drop support for it. I understand there's a presumption that "dev machines aren't servers" and should be more up to date, but in a corporate environment where everything is configured uniformly against a STIG, this is not always the case.

**Comment by iostrym** [comment]
Hello, thanks for all feedback, its sad that on website it is not clearly indicated that RH7 is no longer supported.

Better solution is then to install 1.53 version (2021/01) that is last version that officially support RH7, right ?

About the workaround (gcc10-libstdc++.x86_64 installation), I don't have internet access from our server. Is it possible to install gcc10-libstdc++.x86_64 directly from an RPM downloaded from a website ? or is there a lot of depedency ? there will be no conflict between gcc10-libstdc++.x86_64 and already installed RH7 packages ?

do you know if there are plan to make a new version that will keep compatibility with RH7 ?

**Comment by J-M0** [comment]
@iostrym, the Ghetto Forge libstdc++ doesn't have any dependencies, so you should easily be able to install the rpm on its own. It installs into `/opt` so it shouldn't conflict with anything already on the system.

Direct link: https://mirror.ghettoforge.org/distributions/gf/el/7/gf/x86_64/gcc10-libstdc++-10.2.1-7.gf.el7.x86_64.rpm

**Comment by GitMensch** [comment]
> I (and others) would appreciate an official explanation as to why CentOS 7 is not supported when it has a much longer EOL (End Of Life) than CentOS 8.

Simply because "upstream Electron": https://code.visualstudio.com/updates/v1_53#_electron-11-update

Also see the note above from @Tyriar:

> Sorry but we needed to move forward to support the new version of Electron. Here are your options:
> 
>     * Upgrade your OS
>     * If you're remoting into the machine you can use the remote development extensions or codespaces because this only affects Electron (desktop), not node (remote server)
>     * Try install the compatible binaries as suggested above [#115784 (comment)](https://github.com/microsoft/vscode/issues/115784#issuecomment-774087742), YMMV with this approach
>     * Use an older version of VS Code - not only will you miss out on new updates but also security updates so this isn't recommended

So the question is _not_ "why".
The open question is: will MS provide special RPMs as outlined by @J-M0 at https://github.com/microsoft/vscode/issues/115784#issuecomment-834835385 (which would possible lead to additional bug reports because upstream Electron dropped the support and therefore newer Electron versions may only work if patched) - and yes: I'd personally appreciate that...

**Comment by sifordtj** [comment]
> > I (and others) would appreciate an official explanation as to why CentOS 7 is not supported when it has a much longer EOL (End Of Life) than CentOS 8.
> 
> Simply because "upstream Electron": https://code.visualstudio.com/updates/v1_53#_electron-11-update

I don't expect you to know/see that everything that you have provided I already know.  So many posters in this thread.  I've even already done the workaround and everything is aok (for now).

I guess I worded my question wrong.  Maybe it was better written as "I (and others) would appreciate an official explanation as to why the upstream Electron was considered to be more important than keeping inline with an actual supported operating system (CentOS 7) as opposed to a soon to be unsupported operating system (CentOS 8)."

**Comment by GitMensch** [comment]
I see, then this is a question for another repo - see https://github.com/electron/electron/issues/27480 (and others) that write about "not on the supported list any more".

**Comment by appurist** [comment]
@GitMensch Except that those comments are actually **the part that doesn't make sense**.

I develop Electron 11 and 12 versions of my app on my CentOS 7 target machine and nothing broke compatibility in Electron updates so as far as I can see there's nothing inherent _in the Electron updates_ that require this change. This whole "we had to bump our build image to use Ubuntu-18.04 for x64 linux machines to consume Electron" sounds to me like they just decided to update their OS and noticed things broke and said "$@^#, I guess it's a breaking change." That comment you linked to is actually the part that needs explaining because it doesn't make sense. They broke compatibility for the current LTS version of a major Linux distribution without even blocking automatic updates on that distribution, rendering countless development machines non-functional. In secure environments, it's not possible to just add another lib repo as a "workaround". In some cases, I'm going to need a federal government (military) bureaucrats, working from home due to COVID, to upgrade the laptop they provided, or to bless another library repo.

**Comment by iostrym** [comment]
> 
> 
> @iostrym, the Ghetto Forge libstdc++ doesn't have any dependencies, so you should easily be able to install the rpm on its own. It installs into `/opt` so it shouldn't conflict with anything already on the system.
> 
> Direct link: https://mirror.ghettoforge.org/distributions/gf/el/7/gf/x86_64/gcc10-libstdc++-10.2.1-7.gf.el7.x86_64.rpm

thanks a lot. do we have a feedback for this workaround from people that use massively all function of this tool ? are we well confident about this workaround "working" ?

**Comment by appurist** [comment]
@GitMensch Agreed. I'm not sure what the problem is here since my Electron 11 & 12 apps, and all the others I use, continued to run fine, but only VS Code didn't. I think this means CentOS 7 could be added to that list in the link, yet for some reason VS Code still requires something newer. Perhaps the problem is that VS Code's 18.04 build machine isn't a generic 18.04 but has its libraries upgraded, adding additional dependencies beyond those if it was just building on 18.04. Or something like that. All I know is that it isn't a problem with the Electron binaries as stated.

**Comment by tilleyc** [comment]
> @GitMensch Perhaps the problem is that VS Code's 18.04 build machine isn't a generic 18.04 but has its libraries upgraded, adding additional dependencies beyond those if it was just building on 18.04. Or something like that. All I know is that it isn't a problem with the Electron binaries as stated.

This is it exactly, but they either don't want to admit it, or are unwilling to do so. It's a very easy fix (change to older libs [read: a different build machine] for building the el7 rpms), but in true MS fashion, that's asking too much, so we're all being served flimsy excuses.

I have plenty more I could say, but it's all rather unkind, so I'd rather just leave it at the facts of:

 - this should be a relatively easy fix
 - it requires core Linux knowledge to fix, which seems to be lacking
 - Devs are unable or unwilling to do so and instead want to point fingers and serve excuses

Very poor showing. VScode on Linux is going to see usage plummet, and it deserves to.

**Comment by keegandent** [comment]
@appurist I am in a somewhat similar boat, we are about to try manually compiling and injecting the updated libstdc++.so.6 from the GCC that ships with ubuntu 18.04 to see if that fixes it, but I’m more concerned with what might break going forward.

I was doing this to try and get us away from Eclipse (yes, please kill me) but now I don’t see this becoming the primary editing tool if it’s not even supported for EL7.

It doesn’t bother me that they aren’t supporting a real fix for this, either in VSCode itself or submitting a PR to Electron. What burns my ass is that I *know* if these changes had broken Windows 7 compatibility they wouldn’t have made it into so much as the betas, despite being **5 years older**.

**Comment by deepak1556** [comment]
Thanks @GitMensch and others for helping with the workarounds.

First let me clarify on the toolchain update, when we updated to Electron 11 the v8 headers that come with it were no longer compilable with gcc < 5, these headers are used by native node modules which we bundle. At that time, our build ci was based on ubuntu-14.04, I considered it to be a good time to align vscode ci with same versions electron builds and tests against. Hence today's ci is based on ubuntu-18.04, you can find the config at https://github.com/microsoft/vscode-linux-build-agent/tree/main/bionic-x64

Second, with respect to supported linux versions by Electron. https://github.com/electron/electron/blob/master/docs/tutorial/support.md#linux only specifies the versions that are tested, that does not mean Electron will not work on other versions, it infers issues from outside the list will not be prioritized. Also if chromium runtime works on a particular distro then Electron should work for the most part, and if an application bundles some native component via plugin then the supported distro versions might further dial down. This leads to my final point and the core of this issue.

Most of the native addon that we bundle now requires a higher CXXABI due to the v8 version but it was missed to update the rpm dependency which is what this issue is originally about and it should be fixed. Thanks @J-M0 for the descriptive solution, will give it a try but the earliest I will get to addressing this issue is in July iteration, I am happy to accept PRs in the meantime if anyone wants to help.

**Comment by keegandent** [comment]
@deepak1556 Thanks for the clarification, but can you specify which headers? Also is it that they *only* need a newer compiler version, or is it that they actually use features available only in >=CXXABI_1.3.9? Because if it’s the former, the binaries can likely be fixed (made to work with earlier ABI) with a change to how they are generated. Typically you can use a devtoolset on an older system to use a newer compiler but keep the binaries compatible.

If it’s the latter, maybe we can create a PR in Electron or even v8 to increase the compatibility of those headers and backport the fix to earlier Electron and Code versions.

**Comment by appurist** [comment]
@deepak1556 Okay, if I followed that, it's not an incompatibility with newer Electron versions (which is why I don't see any issue building under CentOS 7 with new Electrons) as much as it's a problem with the fact that VS Code has some native modules that compile against the V8 headers that you get from Electron and those V8 headers correspond to an updated V8 now included in newer Electrons? And those headers V8 now assume a newer GCC than was used previously.

Is there no way to specify a different Electron version when building the native modules? Or different headers themselves? VS Code would run fine on CentOS 7 without the updated GCC, the dependency on the newer CXXABI is not a runtime one but it seems it is a build-time (headers) problem that results in a new runtime dependency. and it's just that by using the newer GCC it's breaking the current latest CentOS LTS, something that will be unavoidably used at many enterprise sites for years to come.

This is enough of a problem for us that we're going to need to reevaluate whether we can still use VS Code at all. Until now, VS Code has been a hands-down favorite and "just works", but with this problem we have a showstopper that means we may need to code with WebStorm or ATOM or good code editors with command-line builds and browser debugging. I really don't want to do that. I love VS Code. But this change, especially during a "minor" 1.53 update, has cost the whole team days of effort just to track down, and cap it at the last working version, as a temporary workaround that won't be permitted longer than a few more weeks. I've been praising Microsoft's major contributions to open source and Linux for some time now, and a lot of that good will gained is going out the window as I start hearing "I told you so". I know it's a pain to resolve, but if you need to do a specific RHEL7/CentOS7 build or something, or build the native modules separately with older headers and tools, or some other workaround like that, it is 100 times better than pushing this ripple out to all your users in a breaking minor update.

**Comment by deepak1556** [comment]
> but can you specify which headers?

https://source.chromium.org/chromium/chromium/src/+/main:v8/include/ and some nodejs public api headers, these will get used via api abstraction library https://github.com/nodejs/nan or https://github.com/nodejs/node-addon-api (which is also ABI stable) depending on what the native module chose to use.

> Also is it that they only need a newer compiler version, or is it that they actually use features available only in >=CXXABI_1.3.9?

Don't remember, have to compile these modules on the older distro to confirm. Will get back on this.

> Typically you can use a devtoolset on an older system to use a newer compiler but keep the binaries compatible.

Yup that would help depending on the result of previous question, but I would like to keep vscode ci distro aligned with what Electron uses for its testing. And this is what Electron currently uses https://github.com/electron/build-images/blob/master/Dockerfile . So it is best if we can come with a solution that does not require using older distro.

> Is there no way to specify a different Electron version when building the native modules? Or different headers themselves

Nope, but if the modules were built with https://github.com/nodejs/node-addon-api then it might be possible to use different headers. 

> I know it's a pain to resolve, but if you need to do a specific RHEL7/CentOS7 build or something, or build the native modules separately with older headers and tools, or some other workaround like that, it is 100 times better than pushing this ripple out to all your users in a breaking minor update.

I understand the frustration caused by this breakage and this issue is kept open to resolve it in a way similar issues don't happen in upcoming electron updates. So the fix will take some time but we will get there.

**Comment by iostrym** [comment]
hello, I don't understand because I retrograde 1.56 version to 1.53 version because it is said here to be the last version compatible with redhat 7.
but recently I discover high CPU usage on 1.53 version for red hat 7 (100% CPU without doing anything in the tool). this is documented here : https://github.com/microsoft/vscode/issues/116092
and this ticket is said to be a dupplicate of this current ticket.
Then my question is the following : what is the version to be used with red hat 7.9 ? 
1.52 version ? sure ?

do I correctly understood that vs code will be corrected this summer to be compatible with red hat 7 ?

**Comment by arthurspa** [comment]
> If you are affected by this issue, let me tell you what I did:
> 
> * Download a recent gcc-tarball from a GNU mirror + compile
> * Take the content of x86_64-pc-linux-gnu/libstdc++-v3/src/.libs and copy to a dir of your choice (eg. ~/lib64/libstdc++-v3)
> * Write a start wrapper for vscode, put your extended LD_LIBRARY_PATH in the wrapper
> * Open vscode with the wrapper -> done.

This worked like a charm!

**Comment by truatpasteurdotfr** [comment]
the same electon excuse has been raised for teams end of support on centos7...

**Comment by iostrym** [comment]
> affected by this issue, let me tell you what I did:
> 
>     * Download a recent gcc-tarball from a GNU mirror + compile
> 
>     * Take the content of x86_64-pc-linux-gnu/libstdc++-v3/src/.libs and copy to a dir of your choice (eg. ~/lib64/libstdc++-v3)
> 
>     * Write a start wrapper for vscode, put your extended LD_LIBRARY_PATH in the wrapper
> 
>     * Open vscode with the wrapper -> done.

this is the workaround to have 1.53 working without CPU problem, right ?

could you provide a zip of the package that we need to place in LD_LIBRARY_PATH  + example of wrapper for vscode ? because i'm not familiar with this :/

**Comment by arthurspa** [comment]
> > affected by this issue, let me tell you what I did:
> > ```
> > * Download a recent gcc-tarball from a GNU mirror + compile
> > 
> > * Take the content of x86_64-pc-linux-gnu/libstdc++-v3/src/.libs and copy to a dir of your choice (eg. ~/lib64/libstdc++-v3)
> > 
> > * Write a start wrapper for vscode, put your extended LD_LIBRARY_PATH in the wrapper
> > 
> > * Open vscode with the wrapper -> done.
> > ```
> 
> this is the workaround to have 1.53 working without CPU problem, right ?
> 
> could you provide a zip of the package that we need to place in LD_LIBRARY_PATH + example of wrapper for vscode ? because i'm not familiar with this :/

I had the library already in my work environment, so I didn't need to download. I just exported the variable before running VSCode, e.g:

```
export LD_LIBRARY_PATH=/app/vbuild/RHEL7-x86_64/gcc/10.2.0/lib64/:$LD_LIBRARY_PATH
code ~/project_folder/
```

And this worked. I think you'd have to compile for your own environment. Here iswhere you can find the source code: http://ftp.gnu.org/gnu/gcc/gcc-10.2.0/

**Comment by kurt-cb** [comment]
IMHO, the correct solution to this problem is to provide a version of vscode that runs on centos 7 native.  This can easily be done using the software collections.

I was able to build the vscode open source using DevToolset-10 C++ compiler toolchain.  This allows it to run native on centos-7
This is the docker file I used, and is on docker hub at kurtgo/vscode-centos7.

Devtoolset-10 is allows the entire thing to be built with compatible with centos7, using the latest gcc compiler.

Steps I used:
docker run -it -v /vsbuild:/home/kurtgo/vsbuild kurtgo/vscode-centos7 bash
# #git pull latest cpython, configure --prefix /user;make -j8 install
# cd /vsbuild;git clone  git@github.com:microsoft/vscode.git;cd vscode
# yarn
# yarn run gulp vscode-linux-x64
# cd ..
# tar cjf centos7-vscode.tar.bz2 vscode-linux-x64

Dockerfile I used to get dependencies:
`
from centos:7

run yum -y install curl; \
    curl -sL https://rpm.nodesource.com/setup_14.x | bash - ; \
    curl --silent --location https://dl.yarnpkg.com/rpm/yarn.repo | tee /etc/yum.repos.d/yarn.repo; \
    rpm --import https://dl.yarnpkg.com/rpm/pubkey.gpg; \
    yum install -y yarn git zlib zlib-devel

run yum -y groupinstall "Development Tools"; \
    yum -y install libxkb*; \
    yum -y install libsecret-devel; \
    yum -y install libX11-devel.x86_64


run yum -y install centos-release-scl; \
    yum -y install devtoolset-10

run yum -y install libssl-devel zlib1g-devel libncurses5-devel \
  libncursesw5-devel libreadline-devel libsqlite3-devel libgdbm-devel \
  libdb5.3-devel libbz2-devel libexpat1-devel liblzma-devel libffi-devel
copy .bashrc /root/.bashrc

run yum -y install ncurses-devel openssl-devel readline-devel sqlite-devel
`

**Comment by keegandent** [comment]
> IMHO, the correct solution to this problem is to provide a version of vscode that runs on centos 7 native. This can easily be done using the softwarecollections project ([www.softwarecollections.org](http://www.softwarecollections.org)).
> 
> I was able to build the vscode open source using DevToolset-10 C++ compiler toolchain. This allows it to run native on centos-7
> This is the docker file I used, and is on docker hub at kurtgo/vscode-centos7.
> 
> Devtoolset-10 is allows the entire thing to be built with compatible with centos7, using the latest gcc compiler.

I would suggest Devtoolset-7 considering the Ubuntu 18.04 CI they are treating as the primary linux target is GCC 7.4, but yes I absolutely agree that having an additional build route to maintain compatibility for Enterprise Linux 7 is not a big ask.

**Comment by kvnlmr** [comment]
We have a n easy workaround on our side with:
- CentOS Linux release 7.9.2009 (Core)
- Visual Studio Code 1.58.2

Instead of installing it with yum package manager, we install it through[ snapcraft.io](url)

First, install Snap on CentOS: [](https://snapcraft.io/docs/installing-snap-on-centos)

```
sudo yum install epel-release -y
sudo yum install snapd -y
sudo systemctl enable --now snapd.socket
sudo ln -s /var/lib/snapd/snap /snap
```

And then, install Visual Studio Code through Snap:

`sudo snap install code --classic`

**Comment by mati-o** [comment]
/verified

I am more than happy to confirm that `1.60.0-insider` version works on CentOS 7! Many thanks!

**Comment by iostrym** [comment]
thanks. just to be sure, now we must also wait that stable version of vscode will get the correction. right ?

**Comment by tazerdev** [comment]
Apologies that I missed the update. I won't be able to verify until Monday, but it's sufficient for me that others have been able to do so. I'll chime in once I am able to confirm.

**Comment by tazerdev** [comment]
/verified

Thanks for fixing this issue.

**Comment by sskamesh** [comment]
/verified

working now with 1.60
